File: 2024-2/Clase26/SegundoProyecto/Persona.java
class Persona{
	private int dni;
	private String nombre;
	private double sueldo;
	private boolean activo;
	private char sexo;

	//Constructores
	public Persona(){
		sueldo = 0.0;
		activo = false;
	}

	public Persona(int dni, String nombre, double sueldo, boolean activo, char sexo){
		this.dni = dni;
		this.nombre = nombre;
		this.sueldo = sueldo;
		this.activo = activo;
		this.sexo = sexo;
	}

	//No necesitamos constructor copia ni destructores

	public void imprimir(){
		System.out.println(dni + " - " + nombre + " - " + sueldo + " - " + sexo + " - " + activo);
	}

	public int get_dni(){
		return dni;
	}

	public String get_nombre(){
		return nombre;
	}

	public double get_sueldo(){
		return sueldo;
	}

	public char get_sexo(){
		return sexo;
	}

	public boolean get_activo(){
		return activo;
	}

	public void set_dni(int dni){
		this.dni = dni;
	}

	public void set_nombre(String nombre){
		this.nombre = nombre;
	}

	public void set_sexo(char sexo){
		this.sexo = sexo;
	}
	
	public void set_sueldo(double sueldo){
		this.sueldo = sueldo;
	}

	public void set_activo(boolean activo){
		this.activo = activo;
	}
}

File: 2025-2/Clase24/STL_MAP/src/Persona.cpp
//
// Created by Erasmo on 5/11/25.
//

#include "Persona.hpp"

Persona::Persona() {
}

Persona::Persona(const Persona &orig) {
}

Persona::~Persona() {
}

void Persona::SetSueldo(double sueldo) {
    this->sueldo = sueldo;
}

double Persona::GetSueldo() const {
    return sueldo;
}

void Persona::SetDni(int dni) {
    this->dni = dni;
}

int Persona::GetDni() const {
    return dni;
}

void Persona::SetNombre(string nombre) {
    this->nombre = nombre;
}

string Persona::GetNombre() const {
    return nombre;
}

void Persona::leer() {
    char buffer[30];
    cin>>dni;
    if(cin.eof())return;
    cin.get();
    cin.getline(buffer, 30, ',');
    nombre = buffer; //Esta sobrecarga ya se encuentra en la clase String
    //SetNombre(buffer);
    cin>>sueldo;
}

void Persona::imprimir() const {
    cout.precision(2);
    cout << fixed;
    cout << right << setw(10) << dni << " " << left << setw(30) << nombre;
    cout << right << setw(10) << sueldo << endl;
}

void operator>>(istream &input, class Persona &p) {
    p.leer();
}

ostream &operator<<(ostream &output, const class Persona &p) {
    p.imprimir();
    return output;
}
